packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.java

```diff
@@ -125,7 +125,13 @@ public static String createFilenameForAsset(AssetEntity asset) {
       // create a filename that's unlikely to collide with any other asset
       return "asset-" + new Date().getTime() + "-" + new Random().nextInt();
     }
-    return asset.key;
+    if (asset.type == null){
+      return asset.key;
+    }
```

Comment:
We don't allow iOS to have a nullable `type`, so there is a bit of divergence.

We do make use of this nullability, in newManifest we don't define the launch asset's extension.